draft-ietf-quic-recovery.md

```diff
@@ -301,63 +302,117 @@ all subsequent ACK frames containing them could be lost. In this case, the
-result is larger than the min_rtt. If the result is smaller than the min_rtt,
-the RTT should be used, but the ack delay field should be ignored.  Ignoring
-ack_delay when it would cause an RTT sample smaller than the current min_rtt
-prevents a peer from manipulating RTT samples below min_rtt at the cost of
-inflated RTT samples in the presence of persistent delayed acks.
-
-A sender takes an RTT sample when an ACK frame is received that acknowledges a
-larger packet number than before.  A sender will take multiple RTT samples per
-RTT when multiple such ACK frames are received within an RTT.  When multiple
-samples are generated within an RTT, the smoothed RTT and RTT variance could
-retain inadequate history, as suggested in {{?RFC6298}}. Changing these
-computations is currently an open research question.
-
-min_rtt is the minimum RTT measured over the connection, prior to adjusting by
-ack delay.  Ignoring ack delay for min RTT prevents intentional or unintentional
-underestimation of min RTT, which in turn prevents underestimating smoothed RTT.
-
-A sender calculates both smoothed RTT (SRTT) and RTT variance (RTTVAR) similar
-to those specified in {{?RFC6298}}.  Note that computing smoothed_rtt does not
-use ack_delay for the first RTT sample, because doing so would result in a
-smoothed_rtt that is smaller than the min_rtt.
-
-On every newly acknowledged ack-eliciting largest acked:
+
+# Computing the round-trip time (RTT) estimate {#compute-rtt}
+
+At a high level, an endpoint measures the time from when a packet was sent to
+when it is acknowledged as a round-trip time (RTT) sample.  The endpoint uses
+RTT samples and peer-reported host delays ({{ack-delay}}, {{generating-acks}}) to
+generate a statistical description of the connection's RTT.  An endpoint
+computes the following three values: the minimum value observed over the
+lifetime of the connection (min_rtt), an exponentially-weighted moving average
+(smoothed_rtt), and the variance in the observed RTT samples (rttvar).
+
+## Generating RTT samples {#latest-rtt}
+
+An endpoint generates an RTT sample on receiving an ACK frame that meets the
+following two conditions:
+
+- the largest acknowledged packet number is higher than any previously seen, and
+
+- at least one of the newly acknowledged packets was ack-eliciting.
+
+The RTT sample, latest_rtt, is generated as the time elapsed since the largest
+acknowledged packet was sent:
+
 ~~~
 latest_rtt = ack_time - send_time_of_largest_acked
 ~~~
 
-First RTT sample:
+An endpoint uses only locally observed times in generating RTT samples and does
+not correct for any host delays reported by the peer ({{ack-delay}}).
+
+An RTT sample MUST NOT be generated for a packet that is not newly acknowledged.
+This avoids multiple RTT samples for the same packet.
+
+An RTT sample MUST NOT be generated for a packet that is not the largest
+acknowledged in the received ACK frame.  A peer reports host delays for only the
```

Comment:
I don't think that you need a "MUST NOT" here.  Your definition of the sample covers this.